Fix type annotations in pandas.core.computation

Part of  #25882

The files in pandas.core.computation have the following errors:
```
pandas/core/computation/ops.py:169: error: Read-only property cannot override read-write property
pandas/core/computation/pytables.py:59: error: Read-only property cannot override read-write property
pandas/core/computation/pytables.py:384: error: Incompatible types in assignment (expression has type "Type[pandas.core.computation.pytables.Constant]", base class "BaseExprVisitor" defined the type as "Type[pandas.core.computation.ops.Constant]")
```

Pull request to follow.

--- a/mypy.ini
+++ b/mypy.ini
@@ -11,15 +11,6 @@ ignore_errors=True
 [mypy-pandas.core.base]
 ignore_errors=True
 
-[mypy-pandas.core.computation.expr]
-ignore_errors=True
-
-[mypy-pandas.core.computation.ops]
-ignore_errors=True
-
-[mypy-pandas.core.computation.pytables]
-ignore_errors=True
-
 [mypy-pandas.core.config_init]
 ignore_errors=True
 

--- a/pandas/core/computation/expr.py
+++ b/pandas/core/computation/expr.py
@@ -7,6 +7,7 @@ from io import StringIO
 import itertools as it
 import operator
 import tokenize
+from typing import Type
 
 import numpy as np
 
@@ -327,7 +328,7 @@ class BaseExprVisitor(ast.NodeVisitor):
     parser : str
     preparser : callable
     """
-    const_type = Constant
+    const_type = Constant  # type: Type[Term]
     term_type = Term
 
     binary_ops = _cmp_ops_syms + _bool_ops_syms + _arith_ops_syms

--- a/pandas/core/computation/ops.py
+++ b/pandas/core/computation/ops.py
@@ -149,10 +149,6 @@ class Term(StringMixin):
     def name(self):
         return self._name
 
-    @name.setter
-    def name(self, new_name):
-        self._name = new_name
-
     @property
     def ndim(self):
         return self._value.ndim
@@ -166,8 +162,7 @@ class Constant(Term):
     def _resolve_name(self):
         return self._name
 
-    # read-only property overwriting read/write
-    @property  # type: ignore
+    @property
     def name(self):
         return self.value
 

--- a/pandas/core/computation/ops.py
+++ b/pandas/core/computation/ops.py
@@ -166,7 +166,8 @@ class Constant(Term):
     def _resolve_name(self):
         return self._name
 
-    @property
+    # read-only property overwriting read/write
+    @property  # type: ignore
     def name(self):
         return self.value
 

--- a/pandas/core/computation/pytables.py
+++ b/pandas/core/computation/pytables.py
@@ -56,7 +56,8 @@ class Term(ops.Term):
         except UndefinedVariableError:
             return self.name
 
-    @property
+    # read-only property overwriting read/write property
+    @property  # type: ignore
     def value(self):
         return self._value
 

